src/appendFrames.test.ts

```diff
@@ -0,0 +1,41 @@
+import { DataFrame, FieldType, MutableDataFrame, Vector } from '@grafana/data';
+
+import { appendMatchingFrames } from './appendFrames';
+
+const v: Vector<number> = {
+  length: 1,
+  get: () => 1,
+  toArray: () => [1],
+};
+
+describe('appendMatchingFrames', () => {
+  it('should do nothing with empty input', () => {
+    expect(appendMatchingFrames([], [])).toEqual([]);
+  });
+
+  it('should return input MutableDataFrames', () => {
```

Comment:
kind of a nit that we can ignore if we don't have good ideas for it:
- in reading this test and the one below I'm not sure I really understand the difference between these two tests. I wonder if we can update the "it" statements to better describe what we're handling or something. Or if these some data frames are mutable should we change them and see that append still handles them the same or something?

Reply:
What about :

```suggestion
  it('should return input when input is a MutableDataFrames array, () => {
```

and next suggestion?

Reply:
sounds good to me